packages/block-editor/src/components/block-actions/index.js

```diff
@@ -43,7 +43,12 @@ export default function BlockActions( {
 
 			return {
 				canRemove: canRemoveBlocks( clientIds ),
-				canInsertBlock: canInsertDefaultBlock || !! directInsertBlock,
+				canInsertBlock: blocks.every( ( block ) => {
+					return (
+						( canInsertDefaultBlock || !! directInsertBlock ) &&
+						canInsertBlockType( block.name, rootClientId )
```

Comment:
This change was needed because "add before" and "add after" options were appearing when it shouldn't be possible to insert blocks.